app/assets/javascripts/components/story/CollapsedStory/index.js

```diff
@@ -46,6 +47,10 @@ export const CollapsedStory = ({ onToggle, story, updateCollapsedStory, project,
       onUpdate={(newAttributes) =>
         updateCollapsedStory(story.id, project.id, newAttributes)}
     />
+    {
+      Story.isSearch(stories.all, story, from) && 
```

Comment:
Receiving all the stories and the from inside a `CollapsedStory` component is a lot of responsibility. Can you elaborate on why this was necessary?